core-rust/state-manager/src/state_manager.rs

```diff
@@ -344,18 +346,12 @@ where
         //========================================================================================
 
         let pending_transaction_base_state = AtState::PendingPreparingVertices {
-            base_committed_state_version: base_transaction_identifiers.state_version,
+            base_committed_state_version: base_accumulator_state.state_version,
         };
 
-        let mut state_tracker = StateTracker::initial(base_transaction_identifiers);
-
-        let ancestor_transactions: Vec<_> = prepare_request
-            .prepared_vertices
-            .into_iter()
-            .flat_map(|v| v.transaction_payloads)
-            .collect();
+        let mut state_tracker = StateTracker::initial(base_accumulator_state);
 
-        for raw_ancestor in ancestor_transactions {
+        for raw_ancestor in prepare_request.prepared_uncommitted_transactions {
```

Comment:
`ancestor` is a cool name, and the rename is trivial - please allow me to avoid self-conflicts though: I will keep a list of renames that should happen after the current series of PRs:

`prepared_uncommitted_transactions -> ancestor_uncommitted_transactions`